my-app/src/PaymentsHistoryTable.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from 'react';
+
+interface PaymentHistoryTableProps {
+  accountId: string;
+}
+
+interface PaymentRecord {
+  date: string;
+  description: string;
+  amount: number;
+  balance: number;
+}
+
+const PaymentHistoryTable: React.FC<PaymentHistoryTableProps> = ({ accountId }) => {
+  const [paymentHistory, setPaymentHistory] = useState<PaymentRecord[]>([]);
+
+  const fetchHistory = async (accountId: string): Promise<void> => {
+    try {
+      const response = await fetch(`https://api.apg.com/creditcards/${accountId}/transactions`);
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
```

Comment:
Errors should be defined, and we should check if the BE is emitting a message here which could get swallowed.